### ISSUE TITLE
Can we support multiple languages ?

### ISSUE THREAD

**Issue by mobilelifeful** [WebDevStudios/wp-search-with-algolia#234]
**Describe the bug**
A site build in English and Spanish,  each post and page are organized in www.example.com/en or www.example.com/es

**To Reproduce**

**Expected behavior**
When we search in English, result in  /en, 
when we search in Spanish, result in  /es.

**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

**Comment by tw2113** [comment]
@mobilelifeful edited your original issue to just the parts provided, and removed the template parts for brevity.

**Comment by tw2113** [comment]
Looks like there's a whole section on this over at https://www.algolia.com/doc/guides/managing-results/optimize-search-results/handling-natural-languages-nlp/

We also have a touch of WPML support enabled as well with the plugin, in case that's what you're using for your internationalization

Beyond that, we're admittedly pretty limited out of box for configuration for handling multiple languages in a given site owner's indexes, but I have to imagine it's possible to accomplish overall with our hooks system to configure as needed.

**Comment by mobilelifeful** [comment]
Using _translatepress_ plugin to help me translate the site,
Also found this plugin search-wpml-algolia which requires to use WPML, although it enables multiple languages search now; but is too expensive for me right now. WPML,

search-wpml-algolia  plugin uses some of WebDevStudios code,  wonder if WebDevStudios can support multiple languages search when _translatepress_ is used.

**Comment by tw2113** [comment]
I suspect it's possible as a whole already, as we have all of the facility to index content, the question is really how to best do that with the least amount of work. Biggest thing is keeping the content potentially separate, and serving the correct version to users, based on chosen language when viewing the site.

**Comment by mobilelifeful** [comment]
no, I tested it, wp-search-with-algolia  work with multiple languages when search-wpml-algolia plugin is installed which also requires  WPML is installed, 
wp-search-with-algolia doesn't work at all when translatepress plugin is installed (even installed with  wp-search-with-algolia is installed)

**Comment by tw2113** [comment]
If we're going to aim for adding out-of-box support, we'd definitely need to test and confirm behavior, and I know all of our WPML integrations are over in https://github.com/WebDevStudios/wp-search-with-algolia/blob/main/includes/class-algolia-compatibility.php which is mostly checking if WPML is enabled, and then doing language switching.

We'd need to explore doing the same, but with TranslatePress instead.

**Comment by mobilelifeful** [comment]
Any option we can enable to provide the log ?

**Comment by asharirfan** [comment]
Adding this issue here just for reference: https://github.com/algolia/algoliasearch-wordpress/issues/567

**Comment by tw2113** [comment]
Looks like TranslatePress does a lot of custom tables to store translations and somehow maps them to the correct spot.

```
wp_trp_dictionary_en_us_en_gb
wp_trp_gettext_en_gb
wp_trp_gettext_en_us
wp_trp_gettext_original_meta
wp_trp_gettext_original_strings
wp_trp_original_meta
wp_trp_original_strings
```
This will make it difficult to index, as compared to other internationalization plugins that rely on unique post entries that are easy to get indexed.
